code/modules/reagents/reagent_containers/ampoules.dm

```diff
@@ -0,0 +1,402 @@
+// Ampoule dot colors:
+/// The dot color used to indicate the presence of a solvent.
+#define AMPOULE_DOT_SOLVENT "white"
+/// The dot color used to indicate the presence of a catalyst.
+#define AMPOULE_DOT_CATALYST "yellow"
+/// The dot color used to indicate the presence of a medicine.
+#define AMPOULE_DOT_MEDICINE "blue"
+/// The dot color used to indicate the presence of a toxin.
+#define AMPOULE_DOT_TOXIN "red"
+/// The dot color used to indicate the presence of a disease.
+#define AMPOULE_DOT_DISEASE "black"
+/// The dot color used to indicate the presence of a radioactive substance.
+#define AMPOULE_DOT_RADIATION "green"
+/// The dot color used to indicate the presence of an explosive.
+#define AMPOULE_DOT_EXPLOSIVE "orange"
+/// The dot color used to indicate the presence of an acid.
+#define AMPOULE_DOT_ACID "violet"
+
+
+/**
+ *
+ *
+ * Behaviors:
+ * - Sealing/Unsealing
+ */
+/obj/item/reagent_containers/glass/ampoule
+	name = "ampoule"
+	desc = "A glass ampoule."
+	icon_state = "ampoule"
+	base_icon_state = "ampoule"
+	inhand_icon_state = "beaker"
+	worn_icon_state = "bottle"
+	custom_materials = list(/datum/material/glass = 25)
+	max_integrity = 10
+	integrity_failure = 0.5
+	force = DAMAGE_PRECISION
+	throwforce = DAMAGE_PRECISION
+	throw_speed = 4
+	throw_range = 7
+	fill_icon_thresholds = list(0, 20, 30, 45, 60, 70, 85)
+	fill_icon_state = "ampoule"
+	volume = 15
+	amount_per_transfer_from_this = 5
+	possible_transfer_amounts = list(5, 10, 15)
+	/// Are we currently sealed?
+	var/sealed = TRUE
+	/// Are we curently broken?
+	var/broken = FALSE
+	/// The gases contained within this ampoule.
+	var/datum/gas_mixture/air_contents
+	/// The gases this ampoule was originally filled with.
+	var/initial_gas_mix
+	/// What color this is marked with.
+	var/dot_color
+	/// What this is labeled as containing.
+	var/label_text
+
+
+/obj/item/reagent_containers/glass/ampoule/Initialize(mapload, vol)
+	. = ..()
+	air_contents = new(volume * 0.1)
```

Comment:
Why?

Reply:
Why am I generating the air there or why is it 1/10 the volume of the reagents?